framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -392,6 +394,71 @@ public AnnotatedTypeMirror getAnnotatedTypeMirror(Element e) {
     return Collections.emptySet();
   }
 
+  /**
+   * Adds annotations from stub files for the corresponding record components (if the given
+   * constructor/method is the canonical constructor or a record accessor). Such transfer is
+   * automatically done by javac usually, but not from stubs.
+   *
+   * @param types a Types instance used for checking type equivalence
+   * @param elt a member. This method does nothing if it's not a method or constructor.
+   * @param memberType the type corresponding to the element elt; side-effected by this method
+   */
+  public void injectRecordComponentType(
+      Types types, Element elt, AnnotatedExecutableType memberType) {
+    if (parsing) {
+      throw new BugInCF("parsing while calling injectRecordComponentType");
+    }
+
+    if (elt.getKind() == ElementKind.METHOD) {
+      if (((ExecutableElement) elt).getParameters().isEmpty()) {
+        String recordName = ElementUtils.getQualifiedName(elt.getEnclosingElement());
+        AnnotationFileParser.RecordStub recordComponentType =
+            annotationFileAnnos.records.get(recordName);
+        if (recordComponentType != null) {
+          // If the record component has an annotation in the stub, the component annotation
+          // replaces any from the same hierarchy on the accessor method, unless there is a specific
+          // annotation on the accessor in the stubs file.
+          // We thus only replace the method annotation with the component annotation
+          // if there is no accessor in the stubs file:
```

Comment:
These comments are inconsistent: "there is a specific annotation" vs "there is an accessor is the stubs file".